src/jit/compiler.cpp

```diff
@@ -11085,3 +11085,231 @@ bool Compiler::killGCRefs(GenTree* tree)
+            // size is not 1, 2, 4 or 8 bytes in size. During fgMorph, the compiler modifies
+            // the IR to comply with the ABI and therefore changes the type of the lclVar
+            // that holds the struct from TYP_STRUCT to TYP_BYREF but it gives us a hint that
+            // this is still a struct by setting the lvIsTemp flag.
+            // The same is true for ARM64 and structs > 16 bytes.
+            // (See Compiler::fgMarkImplicitByRefArgs in Morph.cpp for further detail)
+            // Now, the VM expects a special enum for these type of local vars: VLT_STK_BYREF
+            // to accomodate for this situation.
+            if (varDsc->lvType == TYP_BYREF && varDsc->lvIsTemp)
+            {
+                assert(varDsc->lvIsParam);
+                varLoc.vlType = VLT_STK_BYREF;
+            }
+            else
+#endif // defined(_TARGET_AMD64_) || defined(_TARGET_ARM64_)
+            {
+                varLoc.vlType = VLT_STK;
+            }
+            varLoc.vlStk.vlsBaseReg = baseReg;
+            varLoc.vlStk.vlsOffset  = offset;
+            if (!isFramePointerUsed() && varLoc.vlStk.vlsBaseReg == REG_SPBASE)
+            {
+                varLoc.vlStk.vlsBaseReg = (regNumber)ICorDebugInfo::REGNUM_AMBIENT_SP;
+            }
+            break;
+
+#ifndef _TARGET_64BIT_
+        case TYP_LONG:
+        case TYP_DOUBLE:
+            varLoc.vlType             = VLT_STK2;
+            varLoc.vlStk2.vls2BaseReg = baseReg;
+            varLoc.vlStk2.vls2Offset  = offset;
+            if (!isFramePointerUsed() && varLoc.vlStk2.vls2BaseReg == REG_SPBASE)
+            {
+                varLoc.vlStk2.vls2BaseReg = (regNumber)ICorDebugInfo::REGNUM_AMBIENT_SP;
+            }
+            break;
+#endif // !_TARGET_64BIT_
+
+        default:
+            noway_assert(!"Invalid type");
+    }
+}
+
+//------------------------------------------------------------------------
+// fillRegisterVarLoc: Fill "varLoc" argument indicating the register position of the variable
+// using "LclVarDsc" and "baseReg"/"offset" if it has a part in the stack (x64 bit float or long).
+//
+// Arguments:
+//    varDsc    - a "LclVarDsc *" to the variable it is desired to build the "siVarLoc".
+//    varLoc    - a "siVarLoc &" to fill with the data of the "varDsc".
+//    type      - a "var_types" which indicate the type of the variable.
+//    baseReg   - a "regNumber" use as a base for the offset.
+//    offset    - a signed amount of bytes distance from "baseReg" for the position of the variable.
+//
+// Assumptions:
+//    No new memory is allocated on this method.
+//    No changes on "varDsc" are applied.
+//
+// Notes:
+//    The "varLoc" argument is filled depending of the "type" argument but as a VLT_REG... variation.
+//    A Exception is thrown if type doesn't match with any of the listed on the switch.
+//    "baseReg" and "offset" are used .for not 64 bit and values that are splitted in two parts.
```

Comment:
Something's wrong with this sentence, there's a dot in the middle and the text is strange.

Reply:
Thanks for reading, you are right!